tests/composeapp_test.cc

```diff
@@ -388,25 +418,117 @@ TEST(ComposeApp, handleRemovedApps) {
+  Json::Value installed_target_json;
+  installed_target_json["hashes"]["sha256"] = sha;
+
+  // TODO: consider using test fixtures
+
+  // new App installation
+  {
+    Uptane::Target installed_target("pull", installed_target_json);
+
+    TestClient client("app1", &installed_target);
+
+    Json::Value target_to_install_json{installed_target_json};
+    target_to_install_json["custom"]["docker_compose_apps"]["app1"]["uri"] = "hub.io@factory/app1:sha256:12312312312";
+
+    // We are't doing a fetch, so we have to make this directory so that the
+    // compose_fake script can run:
+    boost::filesystem::create_directories(client.tempdir->Path() / "apps/app1");
+    ASSERT_EQ(data::ResultCode::Numeric::kOk, client.pacman->install({"pull", target_to_install_json}).result_code.num_code);
+
+    std::string output = Utils::readFile(client.tempdir->Path() / "apps/app1/up.log", true);
+    ASSERT_EQ("up --remove-orphans -d", output);
+  }
 
-  TestClient client("app1", &target);
+  // existing App update
+  {
+    installed_target_json["custom"]["docker_compose_apps"]["app1"]["uri"] = "hub.io@factory/app1:sha256:12312312312";
+    Uptane::Target installed_target("pull", installed_target_json);
 
-  // We are't doing a fetch, so we have to make this directory so that the
-  // compose_fake script can run:
-  boost::filesystem::create_directories(client.tempdir->Path() / "apps/app1");
-  ASSERT_EQ(data::ResultCode::Numeric::kOk, client.pacman->install(target).result_code.num_code);
-//  boost::filesystem::remove(client.getRebootSentinel());
-//  auto install_result = client.pacman->finalizeInstall(target);
-  std::string output = Utils::readFile(client.tempdir->Path() / "apps/app1/up.log", true);
-  ASSERT_EQ("up --remove-orphans -d", output);
+    TestClient client("app1", &installed_target);
+
+    Json::Value target_to_install_json{installed_target_json};
+    target_to_install_json["custom"]["docker_compose_apps"]["app1"]["uri"] = "hub.io@factory/app1:sha256:98989898898";
+
+    // We are't doing a fetch, so we have to make this directory so that the
+    // compose_fake script can run:
+    boost::filesystem::create_directories(client.tempdir->Path() / "apps/app1");
+    ASSERT_EQ(data::ResultCode::Numeric::kOk, client.pacman->install({"pull", target_to_install_json}).result_code.num_code);
+
+    std::string output = Utils::readFile(client.tempdir->Path() / "apps/app1/up.log", true);
+    ASSERT_EQ("up --remove-orphans -d", output);
+  }
+
+  // skipping an App update install because already installed
+  {
+    installed_target_json["custom"]["docker_compose_apps"]["app1"]["uri"] = "hub.io@factory/app1:sha256:12312312312";
+    Uptane::Target installed_target("pull", installed_target_json);
+
+    TestClient client("app1", &installed_target);
+
+    Json::Value target_to_install_json{installed_target_json};
+    target_to_install_json["custom"]["docker_compose_apps"]["app1"]["uri"] = "hub.io@factory/app1:sha256:12312312312";
+
```

Comment:
do we need to call pacman->fetch here so we can somehow assert we aren't trying to fetch the app?

Reply:
Yeah, I think, you're right, for some reason, I thought that install() is called on the client not pacman and triggers fetch which obviously is not so, thanks for pointing into it.

Reply:
@doanac I updated the test as well as some other tests. In general, I see lots of things to improve in the test implementation but not sure if it makes sense to invest time & effort in it at this stage (startup) :).

Reply:
I think your approach so far is spot on. We keep chipping away at this as we add/improve things. This is already a tremendous amount of progress.

Reply:
thanks